File: backend/app/schemas/program_standard_mapping.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ProgramStandardMappingBase(BaseModel):
    program_id: uuid.UUID
    standard_id: uuid.UUID
    applicability: str = "required"
    status: str = "active"
    metadata_json: dict | None = None


class ProgramStandardMappingCreate(ProgramStandardMappingBase):
    tenant_id: uuid.UUID


class ProgramStandardMappingUpdate(BaseModel):
    applicability: str | None = None
    status: str | None = None
    metadata_json: dict | None = None


class ProgramStandardMappingResponse(ProgramStandardMappingBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    created_at: datetime
    updated_at: datetime


class ProgramStandardMappingListResponse(BaseModel):
    items: list[ProgramStandardMappingResponse]
    total: int

File: frontend/src/types/programStandardMapping.ts
export type ProgramStandardMappingRecord = {
  id: string;
  tenant_id: string;
  program_id: string;
  standard_id: string;
  applicability: string;
  status: string;
  metadata_json: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
};

export type ProgramStandardMappingListResponse = {
  items: ProgramStandardMappingRecord[];
  total: number;
};